src/libraries/System.IO.Compression/tests/ZipArchive/zip_InvalidParametersAndStrangeFiles.cs

```diff
@@ -907,6 +907,382 @@ public static async Task ZipArchive_InvalidHuffmanData()
+                // the correct value.
+                int updatedLocalVersionToExtract = updatedContents[4];
+                int updatedCentralDirectoryVersionToExtract = updatedContents[101];
+
+                Assert.Equal(20, updatedCentralDirectoryVersionToExtract);
+                Assert.Equal(20, updatedLocalVersionToExtract);
+
+                updatedStream.Seek(0, SeekOrigin.Begin);
+                // Following an update of the ZipArchive, reopen it in read-only mode. Make sure that both entries are correct.
+
+                using (ZipArchive updatedArchive = new ZipArchive(updatedStream, ZipArchiveMode.Read, true))
+                {
+                    Assert.Equal(2, updatedArchive.Entries.Count);
+
+                    ZipArchiveEntry firstEntry = updatedArchive.Entries[0];
+                    ZipArchiveEntry secondEntry = updatedArchive.Entries[1];
+
+                    Assert.Equal("first.bin", firstEntry.Name);
+                    Assert.Equal(10, firstEntry.Length);
+
+                    Assert.Equal("second.bin", secondEntry.Name);
+                    Assert.Equal(4, secondEntry.Length);
+
+                    using (Stream entryStream = firstEntry.Open())
+                    {
+                        byte[] uncompressedBytes = new byte[firstEntry.Length];
+                        int bytesRead = entryStream.Read(uncompressedBytes);
+
+                        Assert.Equal(10, bytesRead);
+
+                        Assert.Equal(new byte[] { 0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09 }, uncompressedBytes);
+                    }
+
+                    using (Stream entryStream = secondEntry.Open())
+                    {
+                        byte[] uncompressedBytes = new byte[secondEntry.Length];
+                        int bytesRead = entryStream.Read(uncompressedBytes);
+
+                        Assert.Equal(4, bytesRead);
+
+                        Assert.Equal(new byte[] { 0x00, 0x01, 0x02, 0x03 }, uncompressedBytes);
+                    }
+                }
+            }
+        }
+
+        [Theory]
+        [InlineData(0x00)]
+        [InlineData(0x14)]
+        public void ZipArchive_InvalidExtraFieldData(byte validVersionToExtract)
+        {
+            byte[] invalidExtraFieldData = s_invalidExtraFieldData.AsSpan().ToArray();
+
+            // Update the sample data's "version needed to extract" field in the local header and the central directory header.
+            // Test this with a valid value (20) to prove that updating a ZipArchive with trailing extra field data won't touch
+            // existing entries. Test it with an invalid value to force ZipArchive to write the corrected header (preserving the
+            // trailing data.)
+            invalidExtraFieldData[4] = validVersionToExtract;
+            invalidExtraFieldData[60] = validVersionToExtract;
+
+            using (MemoryStream updatedStream = new MemoryStream())
```

Comment:
super nit: since these memory streams get disposed at the very end of the method can use the new using format and you can save a bunch of indentation in the next lines by removing the curly brackets that wrap all that code. The memory stream will be disposed at the end of the current context.

```suggestion
            using MemoryStream updatedStream = new MemoryStream();
```